src/Compilers/CSharp/Portable/Binder/ForLoopBinder.cs

```diff
@@ -70,8 +77,15 @@ private BoundForStatement BindForParts(ForStatementSyntax node, Binder originalB
             if (node.Declaration != null)
             {
                 Debug.Assert(node.Initializers.Count == 0);
-                ImmutableArray<BoundLocalDeclaration> unused;
-                initializer = originalBinder.BindForOrUsingOrFixedDeclarations(node.Declaration, LocalDeclarationKind.ForInitializerVariable, diagnostics, out unused);
+                if (node.Declaration.Deconstruction == null)
```

Comment:
Did we consider adding a property `IsDeconstructionDeclaration`  or such?  The `null` check seems indirect.